In [1]:
import pandas as pd
import numpy as np
import plotly.express as px

Why I pick the dataset? & General Question:

As one of the countries with highest income, I am curious about the income distribution of United States. Especially whether the income stats is related with location.

In [2]:
df=pd.read_csv("https://github.com/upb3y/bdi475/blob/main/kaggle_income.csv?raw=true", encoding='latin-1')

In [3]:
df.head()

id  State_Code State_Name State_ab          County            City  \
0  1011000           1    Alabama       AL   Mobile County       Chickasaw   
1  1011010           1    Alabama       AL  Barbour County      Louisville   
2  1011020           1    Alabama       AL   Shelby County      Columbiana   
3  1011030           1    Alabama       AL   Mobile County         Satsuma   
4  1011040           1    Alabama       AL   Mobile County  Dauphin Island   

             Place  Type Primary  Zip_Code Area_Code     ALand     AWater  \
0   Chickasaw city  City   place     36611       251  10894952     909156   
1        Clio city  City   place     36048       334  26070325      23254   
2  Columbiana city  City   place     35051       205  44835274     261034   
3      Creola city  City   place     36572       251  36878729    2374530   
4   Dauphin Island  Town   place     36528       251  16204185  413605152   

         Lat        Lon   Mean  Median  Stdev        sum_w  
0  30.771450 -88.079697  38773   30506  33101  1638.260513  
1  31.708516 -85.611039  37725   19528  43789   258.017685  
2  33.191452 -86.615618  54606   31930  57348   926.031000  
3  30.874343 -88.009442  63919   52814  47707   378.114619  
4  30.250913 -88.171268  77948   67225  54270   282.320328

In [4]:
df.tail()

id  State_Code   State_Name State_ab              County  \
32521   720296          72  Puerto Rico       PR  Adjuntas Municipio   
32522  7202966          72  Puerto Rico       PR  Adjuntas Municipio   
32523  7202976          72  Puerto Rico       PR  Adjuntas Municipio   
32524  7202986          72  Puerto Rico       PR  Adjuntas Municipio   
32525  7202996          72  Puerto Rico       PR  Adjuntas Municipio   

            City     Place   Type Primary  Zip_Code Area_Code     ALand  \
32521   Guaynabo  Adjuntas  Track   Track       970       787    589417   
32522     Aguada  Adjuntas  Track   Track       602       787   1801613   
32523     Aguada  Adjuntas  Track   Track       602       787  11031227   
32524     Aguada  Adjuntas  Track   Track       602       787         0   
32525  Aguadilla  Adjuntas  Track   Track       603       787   6476604   

         AWater        Lat        Lon   Mean  Median  Stdev        sum_w  
32521      1691  18.397925 -66.130633  30649   13729  37977  1321.278082  
32522    795887  18.385424 -67.203310  15520    9923  15541   238.813450  
32523         0  18.356565 -67.180686  41933   34054  31539   313.551070  
32524  33597561  18.412041 -67.213413      0       0      0     0.000000  
32525   2717115  18.478094 -67.160453  28049   20229  33333   512.884803

In [5]:
df.sample(5)

id  State_Code  State_Name State_ab           County  \
26502  47021280          47   Tennessee       TN  Anderson County   
15757  27024853          27   Minnesota       MN    Aitkin County   
2052   60212819           6  California       CA   Alameda County   
16321  29021154          29    Missouri       MO     Adair County   
13235  24024437          24    Maryland       MD  Allegany County   

              City            Place   Type Primary  Zip_Code Area_Code  \
26502     Big Rock     Clinton city  Track   Track     37023       931   
15757     Shakopee      Aitkin city  Track   Track     55379       952   
2052   Los Angeles     Alameda city  Track   Track     90032       323   
16321     St Louis  Kirksville city  Track   Track     63129       314   
13235  Deal Island   Little Orleans  Track   Track     21821       410   

           ALand     AWater        Lat         Lon    Mean  Median  Stdev  \
26502  101257990     304500  36.595375  -87.687171       0       0      0   
15757   30396904     952721  44.755292  -93.488962  121844  114103  64204   
2052      439362          0  34.085173 -118.173901   43352   35055  37580   
16321    8511872     128094  38.475773  -90.338565   89125   77504  50684   
13235          0  380178087  38.011948  -76.067266       0       0      0   

             sum_w  
26502     0.000000  
15757   746.432867  
2052    347.650597  
16321  1314.416729  
13235     0.000000

In [6]:
df.shape

(32526, 19)

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32526 entries, 0 to 32525
Data columns (total 19 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id          32526 non-null  int64  
 1   State_Code  32526 non-null  int64  
 2   State_Name  32526 non-null  object 
 3   State_ab    32526 non-null  object 
 4   County      32526 non-null  object 
 5   City        32526 non-null  object 
 6   Place       32526 non-null  object 
 7   Type        32526 non-null  object 
 8   Primary     32526 non-null  object 
 9   Zip_Code    32526 non-null  int64  
 10  Area_Code   32526 non-null  object 
 11  ALand       32526 non-null  int64  
 12  AWater      32526 non-null  int64  
 13  Lat         32526 non-null  float64
 14  Lon         32526 non-null  float64
 15  Mean        32526 non-null  int64  
 16  Median      32526 non-null  int64  
 17  Stdev       32526 non-null  int64  
 18  sum_w       32526 non-null  float64
dtypes: float64(3), int64(8), 

Question 1:
Find the ten places with largest mean income. 
Find the position of Champaign County in the data set.

In [8]:
df_desc_by_mean_income = df.sort_values('Mean',ascending=False,ignore_index=True)
df_desc_by_mean_income.head(10)

id  State_Code    State_Name State_ab  \
0     201700           2        Alaska       AK   
1   60224419           6    California       CA   
2   42012156          42  Pennsylvania       PA   
3     101920           1       Alabama       AL   
4   34024395          34    New Jersey       NJ   
5  360219824          36      New York       NY   
6  120210380          12       Florida       FL   
7  480210018          48         Texas       TX   
8   51024094          51      Virginia       VA   
9   60220539           6    California       CA   

                            County              City               Place  \
0  Southeast Fairbanks Census Area    Delta Junction          Whitestone   
1                   Alameda County         San Diego        Alameda city   
2                   Chester County      West Chester  Cheyney University   
3                 Jefferson County         Odenville                Argo   
4                  Atlantic County       Short Hills        Absecon city   
5                    Albany County        Bronxville         Albany city   
6                   Alachua County       Miami Beach    Gainesville city   
7                  Anderson County            Dallas             Elkhart   
8                  Accomack County          Leesburg        Chincoteague   
9                   Alameda County  Huntington Beach        Alameda city   

    Type Primary  Zip_Code Area_Code     ALand   AWater        Lat  \
0    CDP   place     99737       907  18298887        0  64.152838   
1  Track   Track     92110       619   1961071    81282  32.737719   
2    CDP   place     19382       484    604077        0  39.933520   
3   Town   place     35120       205  27893577   150331  33.691576   
4  Track   Track      7078       973   9094477  1302363  40.739665   
5  Track   Track     10708       914   1805685        0  40.938765   
6  Track   Track     33139       305   1420761  3174610  25.756717   
7  Track   Track     75225       214   2883569        0  32.858013   
8  Track   Track     20175       571   8876960    48623  39.096570   
9  Track   Track     92648       714   3060683   897184  33.679397   

          Lon    Mean  Median  Stdev      sum_w  
0 -145.906385  242857  300000  25317   0.866493  
1 -117.197744  242857  300000  25317   0.733187  
2  -75.530306  242857  300000  25317   0.599880  
3  -86.503766  242857  300000  25317   1.133107  
4  -74.342521  216503  300000  56951  22.729221  
5  -73.823654  209392  300000  67438  59.393102  
6  -80.140066  207128  233504  78050  21.595832  
7  -96.799359  206380  300000  63968  39.826347  
8  -77.595294  205835  300000  61701  62.158033  
9 -118.020316  203910  224262  68202  82.853472

In [9]:
df_desc_by_mean_income[df_desc_by_mean_income['County']=='Champaign County']

id  State_Code State_Name State_ab            County     City  \
12118  17012787          17   Illinois       IL  Champaign County  Gifford   

         Place     Type Primary  Zip_Code Area_Code    ALand  AWater  \
12118  Gifford  Village   place     61847       217  1167265       0   

             Lat        Lon   Mean  Median  Stdev       sum_w  
12118  40.307844 -88.021335  69910   58969  46551  187.224133

In [10]:
len(df_desc_by_mean_income)

32526

Question 2: find the top 5 mean income state.

In [11]:
df_by_state = df.copy()
df_by_state['Total Household income'] = df_by_state['sum_w']*df_by_state['Mean']
df_by_state.drop(columns=['id','State_Code','Zip_Code','ALand','AWater','Lon','Lat','Mean','Median','Stdev','County','City','Place','Type','Primary','Area_Code'],inplace=True)
df_by_state = df_by_state.groupby(by='State_Name',as_index=False).agg('sum')
df_by_state['mean']=df_by_state['Total Household income']/df_by_state['sum_w']

In [12]:
df_by_state.sort_values('mean',ascending=False,inplace=True)
df_by_state.head(5)

State_Name          sum_w  Total Household income          mean
8   District of Columbia  156245.519491            1.399367e+10  89562.034400
11                Hawaii   67282.514576            5.645360e+09  83905.307572
6            Connecticut  187943.617152            1.521578e+10  80959.271520
20              Maryland  254189.171805            2.041937e+10  80331.406927
1                 Alaska   29203.482215            2.342456e+09  80211.532896

Question 3: mean inome distribution by location

In [13]:
fig = px.histogram(df, x="Mean", title='income distribution histogtram')
fig.show()

In [14]:
df_only_states = df[df['State_Name']=='Illinois']
fig = px.box(df_only_states, x="Mean", title='income distribution box graph of illinois')
fig.show()

Communicate my findings:

In [15]:
fig = px.scatter(df, x="ALand",y="Mean", title='relation between land and mean income')
fig.show()

In [16]:
fig = px.scatter(df, x="AWater",y="Mean",title='relation between water usage and mean income')
fig.show()

In [17]:

fig = px.scatter(df_by_state, x="Total Household income",y="mean",title='relation between total income and mean income')
fig.show()

In [18]:
fig = px.pie(df_by_state, values='Total Household income', names='State_Name', title = 'total income of united states')
fig.update_traces(textposition='inside')
fig.update_layout(uniformtext_minsize=12, uniformtext_mode='hide')

fig.show()

As we can see in the above graphs, the mean income is not related with size of the land and water usage. The US mean income by states largely falls in 50k to 60k. The states with largest total income (california and illinois) doesn't have a highest mean income among the states. DC has the highest average income of 89562.